### DISCUSSION TITLE
Alt Kit Keyboard View: "Just Show Me The Sounds"

### DISCUSSION THREAD

**Discussion by scheelio** [SynthstromAudible/DelugeFirmware#629] [Ideas]
While I appreciate the 16 levels view that was added, I think another useful Keyboard View option for kits would be to simply allow the user to have access to more than 8 sounds at a time. My main use case would be for flipping samples -- being able to just trigger slice 1 and then slice 18 without having to scroll would be a big workflow improvement. 

### **Goal:**

When hitting the Keyboard View button in a kit, the main pads would show the first 128 sounds loaded into that kit and allow the user to trigger those sounds from the main grid pads.

### **Implementation Thoughts:** 

- Access options  
  - Could be added as a second option when hitting the Keyboard View button 
    - 1st press would be 16 Levels view, second press would be All Sounds view
  - Shift + Keyboard View button could toggle between the two modes
  - Use the Launch/Audition pads to toggle between views (kind of like how they are used to change the behavior in the grid view that was added)
  - Otherwise, a user setting to choose mode
- Layout of the sounds on the pads could be grouped in sets of 16 (for finger drumming)
  - Start at the bottom left, sounds 1-4 would be placed on pads 1-4 of the bottom row, sounds 5-8 would be placed on pad 1-4 of the next row up, etc.; once 16 sounds are hit, move to pads 5-8 and follow the same, then repeat the same for the top four rows of pads (starting at sound 65)
  - This could potentially cause confusion with the 16 Levels view, however
- Just filling in the sounds starting with the bottom left pad from left to right, moving up a row when hitting sound 17, would still be useful
- Recording while "playing" the pads would hopefully be supported
- If a kit has more than 128 sounds, using vertical encoder to scroll to sounds beyond 128 would hopefully be supported

That's all I can think of right now. I think this would be a nice addition for those of use that like to slice samples for flipping, or for those that like to have a 'main' drum kit that has a lot of sounds in it.

**Comment by ok-reza** [comment]
you can hold shift+turn the <> encoder to zoom in or out of kit keyboard view to have a sample per pad. (personally i think the shortcut should be press and turn <> encoder (similarly to how we zoom in and out of clip view). there currently isn't a nice solve for easily rearranging pads while in kit keyboard view yet though.

additionally, all samples will be triggered with velocity of 127 (max). i think it would be good to have an option to respect the default velocity the deluge has when pads view is 1x1.

**Reply by scheelio** [reply]
Ahh, I did not catch the zoom out function of the 16 Levels view. I actually did try the push-and-turn like you mentioned, so I didn't think it existed. Good to know. 

I agree that max velocity is not ideal and would prefer it to reset back to default when zoomed all the way out, too. 

Even though I'm not a finger drummer, I do like the 4x4 grid when triggering samples. So maybe that kind of organization along with the velocity change would enough to differentiate it and justify a new view? Will see what others think.
